gateway/types.go

```diff
@@ -0,0 +1,125 @@
+package gateway
```

Comment:
Why is it called gateway? The content of the package is different types. So it would be good to name it `types`.

Reply:
If they are gateway actor types, then it is fine with me.